packages/react/src/components/DashboardEditor/DashboardEditorCardRenderer.jsx

```diff
@@ -164,65 +126,225 @@ const renderCustomCard = (props) => {
+  );
+
+  return (
+    <DashboardEditorCardRenderer {...others} dataItems={dataItemsForCard} style={cachedStyle}>
+      {
+        cachedChildren // this is a performance enhancement because the children were always being recreated
+      }
+    </DashboardEditorCardRenderer>
+  );
+};
+
+CachedEditorCardRenderer.defaultProps = {
+  style: {},
+};
+
+const shouldComponentSkipUpdate = (prevProps, nextProps) => {
+  return isEqual(prevProps, nextProps);
+};
+
+const availableActions = { clone: true, delete: true };
 /**
  * Returns a Card component for preview in the dashboard
  * @param {Array} dataItems list of dataItems available to the card
  * @param {Object} availableDimensions collection of dimensions where the key is the
  * dimension and the value is a list of values for that dimension
  * @returns {Node}
  */
-const DashboardEditorCardRenderer = ({ dataItems, availableDimensions, ...others }) => {
-  const [dynamicHotspots, setDynamicHotspots] = useState([]);
+const DashboardEditorCardRenderer = React.memo(
+  ({
+    isSelected,
+    dataItems,
+    availableDimensions,
+    i18n,
+    onRemove,
+    onDuplicate,
+    onCardChange,
+    setSelectedCardId,
+    renderIconByName,
+    onShowImageGallery,
+    onValidateUploadedImage,
+    baseClassName,
+    renderCardPreview,
+    style,
+    ...cardConfig
+  }) => {
+    // We have to keep track of our dynamic hotspots here
+    const [dynamicHotspots, setDynamicHotspots] = useState([]);
+
+    const handleOnCardAction = useCallback(
+      (id, actionId, payload) => {
+        if (actionId === CARD_ACTIONS.CLONE_CARD) {
+          onDuplicate(id);
+        } else if (actionId === CARD_ACTIONS.DELETE_CARD) {
+          onRemove(id);
+        } else if (actionId === CARD_ACTIONS.ON_CARD_CHANGE) {
+          onCardChange(update(cardConfig, payload));
+        }
+      },
+      [cardConfig, onCardChange, onDuplicate, onRemove]
+    );
 
-  useEffect(() => {
-    const originalDynamicHotspot = others.values?.hotspots?.find(
-      (hotspot) => hotspot.type === 'dynamic'
+    const handleOnCardKeyDown = useCallback(
+      (e) => {
+        handleKeyDown(e, setSelectedCardId, cardConfig.id);
+      },
+      [cardConfig.id, setSelectedCardId]
     );
-    // if we have dynamic hotspots and a way to fetch them, fetch them
-    if (originalDynamicHotspot && others.onFetchDynamicDemoHotspots) {
-      others.onFetchDynamicDemoHotspots().then((hotspots) =>
-        setDynamicHotspots(
-          hotspots.map((hotspot) => ({
-            ...hotspot,
-            ...omit(originalDynamicHotspot, 'type', 'x', 'y'),
-          }))
-        )
+
+    const handleOnCardMouseDown = useCallback(
+      (e) => {
+        handleOnClick(setSelectedCardId, cardConfig.id);
+        cardConfig.onMouseDown(e);
+      },
+      // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Whats being disabled here? I think all of these should be able to be memoized

Reply:
believe it or not, it's a bug with the rule, I don't want the callback to trigger for any change of the cardConfig, only those two fields, but if I don't disable, it will switch it to 'cardConfig'